_posts/05-06-01-Internationalization-and-Localization.md

```diff
@@ -0,0 +1,348 @@
+throughout all your `.po` files, meaning other languages will have the same format and the same `msgid` fields, but
+translated `msgstr` lines.
+
+Talking about translation keys, there are two main "schools" here:
+
+1. _`msgid` as a real sentence_.  
+    The main advantage are: 
+    - if there's pieces of the software untranslated in any given language, the key displayed will still maintain some
+    meaning. Example: if you happen to translate by heart from English to Spanish but needs help to translate to French,
+    you might publish the new page with missing French sentences, and parts of the website would be displayed in English
+    instead;
+    - it's much easier for the translator to understand what's going on and make a proper translation based on the
+    `msgid`;
+    - it gives you "free" l10n for one language - the source one;
+    - The only disadvantage: if you need to change the actual text, you would need to replace the same `msgid`
+    across several language files.
+
+2. _`msgid` as a unique, structured key_. It would describe the sentence role in the application in a structured way,
+including the template or part where the string is located instead of its content.
+    - it's a great way to have the code organized, separating the text content from the template logic.
+    - however, that could bring problems to the translator that would miss the context. A source language file would be
+    needed as a basis for other translations. Example: the developer would ideally have an `en.po` file, that
+    translators would read to understand what to write in `fr.po` for instance.
+    - missing translations would display meaningless keys on screen (`TOP_MENU_WELCOME` instead of `Hello there, User!`
+    on the said untranslated French page). That's good as would force translation to be complete before publishing -
+    but bad as translation issues would be really awful in the interface.
+
+The [Gettext manual][manual] favors the first approach, as in general it's easier for translators and users in
+case of trouble. That's how we will be working here as well.
+
+### Everyday usage
+In a common application, you would use some Gettext functions while writing static text in your pages. Those sentences
+would then appear in `.po` files, get translated, compiled into `.mo` files and then, used by Gettext when rendering
+the actual interface. Given that, let's tie together what we have discussed so far in a a step-by-step example:
+
+#### 1. A sample template file, including some different gettext calls
+{% highlight php %}
+<?php include 'i18n_setup.php' ?>
+<div id="header">
+    <h1><?=sprintf(gettext('Welcome, %s!'), $name)?></h1>
+    <!-- code indented this way only for legibility here -->
+    <?php if ($unread): ?>
+        <h2><?=sprintf(
+            ngettext('Only one unread message',
+                     '%d unread messages',
+                     $unread),
+            $unread)?>
+        </h2>
+    <? endif ?>
```

Comment:
Given that you used `<?php` as your open tag earlier, I recommend also using it on your `<? endif ?>` line